src/CommunityToolkit.Maui/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,177 @@
+		where TView : NavigableElement
+		where TViewModel : class, INotifyPropertyChanged
+	{
+		RegisterShellRoute<TView>(route, factory);
+		return services.AddTransient<TView, TViewModel>();
+	}
+
+	/// <summary>
+	/// Adds a Page or View of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Singleton"/> lifetime.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="BindableObject"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of <typeparamref name="TViewModel" /> 
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddSingleton<TView, TViewModel>(this IServiceCollection services)
+		where TView : BindableObject
+		where TViewModel : class, INotifyPropertyChanged
+	{
+		return services
+			.AddSingleton<TViewModel>()
+			.AddSingleton<TView>();
+	}
+
+	/// <summary>
+	/// Adds a Page of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Singleton"/> lifetime
+	/// and registers a MAUI Shell route in <see cref="Routing"/> using the value of 
+	/// <paramref name="route"/> as the route.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="NavigableElement"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <param name="route">Route at which this page will be registered within Shell routing. 
+	/// Defaults to type name of <typeparamref name="TView" />.</param>
+	/// <param name="factory">RouteFactory to be used while creating the <see cref="NavigableElement"/> 
+	/// for the route. Defaults to TypeRouteFactory.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of <typeparamref name="TViewModel" /> 
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddSingletonWithShellRoute<TView, TViewModel>(this IServiceCollection services, string? route = null, RouteFactory? factory = null)
```

Comment:
Let's remove `string? route = null`, changing it to `string route`.

I'm very concerned that people who don't understand Shell will use our default route implementation, then blame us for it not being what they wanted it to be.

I prefer that we force developers to be thoughtful and include a route, just like `Routing.RegisterRoute` forces the developer to input a route.

If .NET MAUI's Routing.RegisterRoute included a default route, I'd be cool with us leveraging that default, but it doesn't.

```suggestion
	public static IServiceCollection AddSingletonWithShellRoute<TView, TViewModel>(this IServiceCollection services, string route, RouteFactory? factory = null)
```

Reply:
This change will be present in my update to the PR. 😞 I don't like it, but I understand.